### PR TITLE
Component story format v3

### PR DESCRIPTION
**Pull request by ahuth**
### Summary:

Convert stories to [component story format (CSF) v3](https://storybook.js.org/blog/component-story-format-3-0/).

### Test Plan:

- CI

### CODE REVIEW

File: packages/components/src/Banner/Banner.stories.tsx [hunk 1]
```diff
@@ -15,186 +15,203 @@ export default {
       },
     },
   },
+  args: {
+    content:
+      "Summit Learning has a full-time team dedicated to constantly improving our curriculum. To see the updates, click into the course.",
+    heading:
+      "New curriculum updates are available for one or more of your courses.",
+  },
```
Comment (ahuth) [inline packages/components/src/Banner/Banner.stories.tsx:23]: Top level args are inherited by each story.

We could always do this, but we used to lint against it due to limitations of `prepareStory`.

File: packages/components/src/Button/button.stories.tsx [hunk 2]
```diff
@@ -69,17 +68,21 @@ const gridParameters = {
   },
 };
 
-export const AllVariants = () => getAllVariantsWithStates("button", "Button");
-AllVariants.parameters = gridParameters;
+export const AllVariants = {
```
Comment (ahuth) [inline packages/components/src/Button/button.stories.tsx:71]: Generally I think we should explicitly type stories - i.e.

```ts
export const AllVariants: StoryObj<Args> = {
```

But for these stories that don't use or set args, it doesn't really matter.

File: packages/components/src/Heading/Heading.stories.tsx [hunk 3]
```diff
@@ -9,48 +9,53 @@ export default {
...
-
-export const Heading1 = Template.bind(null);
-Heading1.args = {
-  size: "h1",
-  children: "Heading 1 24/32",
+export const Heading1: StoryObj<Args> = {
+  args: {
+    size: "h1",
+    children: "Heading 1 24/32",
+  },
```
Comment (ahuth) [inline packages/components/src/Heading/Heading.stories.tsx:16]: Notice that none of the stories actually define a `render` function...

For any story that does nothing but apply `args` to the component - i.e.

```tsx
(args) => <Heading {...args />;
```

we can omit it and Storybook will do that for us.

File: packages/components/src/SvgIcon/SvgIcon.stories.tsx [hunk 4]
```diff
@@ -41,119 +41,139 @@ type Args = React.ComponentProps<typeof SvgIcon> & {
...
 
-const Template: Story<Args> = ({ icon, color, ...rest }) => {
-  const Icon = allIcons[icon];
-  const computedColor = color && ColorTokens[color];
-  return <Icon {...rest} color={computedColor} />;
-};
-
-const defaultArgs = {
-  icon: Object.keys(allIcons)[0] as keyof typeof allIcons,
-  role: "decorative" as const,
```
Comment (ahuth) [inline packages/components/src/SvgIcon/SvgIcon.stories.tsx:52]: Notice that `role` here isn't a valid prop (anymore), but TS failed to catch it.

Looks like object stories get us better types, and TS would not catch this.

File: packages/components/src/Button/button.stories.tsx (hunk 2 above)
Comment (anniehu4) [reply packages/components/src/Button/button.stories.tsx:71]: Could you explain why we should explicitly type every story?

**Review by anniehu4 — APPROVED** [review]
niice love how much more condensed it is

File: packages/components/src/Button/button.stories.tsx (hunk 2 above)
Comment (ahuth) [reply packages/components/src/Button/button.stories.tsx:71]: Mostly so it forces us to get the `args` right (if we are using any).

Also I think `composesStories` requires them to be annotated to infer types in tests.

Comment (ahuth) [reply packages/components/src/Button/button.stories.tsx:71]: Maybe we should think about a lint rule for that 🤔

**Comment by ahuth** [comment]
Thanks!

**Pull request merged**